shared/actions/fs.js

```diff
@@ -212,29 +212,35 @@ function* pollSyncStatusUntilDone(): Saga.SagaGenerator<any, any> {
     return
   }
   polling = true
+  let syncingSet = false
```

Comment:
Why do you need this variable? With my suggested flow, it all happens linearly so it doesn't need to go into the `finally` block. Is it to protect against Saga cancelation?

Reply:
Yeah I was worried about the error path. But I guess we could just update it to false in `finally` block.

Reply:
Does the `finally` even get executed with a canceled generator? What if the `finally` has `yield`s?

Reply:
And yeah, I think hard setting it in the `finally` is harmless.

Reply:
> Does the finally even get executed with a canceled generator?

Yeah I think so: https://redux-saga.js.org/docs/advanced/TaskCancellation.html

Reply:
Perfect I was looking for exactly that kind of example but couldn't find it anywhere. Thanks!